packages/record-data/addon/-private/ts-interfaces/relationship-record-data.ts

```diff
@@ -21,6 +22,7 @@ export interface RelationshipRecordData extends RecordData {
   isNew(): boolean;
   modelName: string;
   storeWrapper: RecordDataStoreWrapper;
+  identifier: StableRecordIdentifier;
```

Comment:
Would you consider this PR more than test coverage?  I see we are adding a new `recordDataFor` function declaration.  However, I'm unclear as to its usage (currently, unless you have more planned).  Is the "future" plans for building up a weak map captured in the TODO above?

Reply:
It's currently used by the tests and will be used extensively by the new relationship layer. The TODO is for our InternalModel burndown: this iteration logic needs to move into a combination of the graph / store as currently the only thing causing RecordData's to not be retained is that they are only entangled with InternalModel and so when we release an InternalModel we by result release the RecordData.

Reply:
(The tests need this because they need an abstract way to peek for whether there is a RecordData for an identifier without accidentally rematerializing it or cancelling destruction)